lib/internal/url.js

```diff
@@ -647,6 +647,35 @@ function getObjectFromParams(array) {
   return obj;
 }
 
+// Mainly to mitigate func-name-matching ESLint rule
+function defineIDLClass(proto, classStr, obj) {
```

Comment:
That would likely be a bit overkill at the moment. This is the only IDL defined API we have currently. Eventually if we go down that road more then we can consider this.